asn1sccserviceprovider.cpp

```diff
@@ -105,15 +105,23 @@ QStringList Asn1SccServiceProvider::additionalArguments() const
     return arguments;
 }
 
-QJsonDocument Asn1SccServiceProvider::buildAstRequestData(const QString &inputData, const QFileInfo &fileInfo) const
+QJsonDocument Asn1SccServiceProvider::buildAstRequestData(const QHash<QString, DocumentSourceInfo> &documents) const
 {
-    QJsonObject asnFileData;
+    QJsonArray documentArray;
 
-    asnFileData["Name"] = fileInfo.fileName().toStdString().c_str();
-    asnFileData["Contents"] = inputData.toStdString().c_str();
+    QHashIterator<QString, DocumentSourceInfo> iter(documents);
+    while (iter.hasNext()) {
+        iter.next();
+
+        QJsonObject asnFileData;
+        asnFileData["Name"] = iter.value().getName().toStdString().c_str();
```

Comment:
(the `toStdString().c_str()` idiom was used in Qt4 to migrate between UTF-8 world and "locale" world, but I think it's not the case here)